astroquery/mast/missions.py

```diff
@@ -0,0 +1,187 @@
+
+        Returns
+        -------
+        response : `~astropy.table.Table`
+        """
+
+        return self._service_api_connection._parse_result(response, verbose, data_key='results')
+
+    @class_or_instance
+    def query_region_async(self, coordinates, radius=3*u.arcmin, **kwargs):
+        """
+        Given a sky position and radius, returns a list of matching dataset IDs.
+
+        Parameters
+        ----------
+        coordinates : str or `~astropy.coordinates` object
+            The target around which to search. It may be specified as a
+            string or as the appropriate `~astropy.coordinates` object.
+        radius : str or `~astropy.units.Quantity` object, optional
+            Default 3 degrees.
+            The string must be parsable by `~astropy.coordinates.Angle`. The
+            appropriate `~astropy.units.Quantity` object from
+            `~astropy.units` may also be used. Defaults to 3 arcminutes.
+        **kwargs
+            Other mission-specific keyword args.
+            These can be found at the following link
+            https://mast.stsci.edu/search/docs/#/Hubble%20Search/post_search_hst_api_v0_1_search_post
+            For example one can specify the output columns(select_cols) or use other filters(conditions)
+
+        Returns
+        -------
+        response : list of `~requests.Response`
+        """
+
+        # Put coordinates and radius into consistant format
+        coordinates = commons.parse_coordinates(coordinates)
+
+        # if radius is just a number we assume degrees
+        radius = coord.Angle(radius, u.arcmin)
+
+        # basic params
+        params = {'target': [f"{coordinates.ra.deg} {coordinates.dec.deg}"],
+                  'radius': radius.arcmin,
+                  'radius_units': 'arcminutes'}
+
+        params['conditions'] = []
+        # adding additional user specified parameters
+        for prop, value in kwargs.items():
+            if prop not in self._search_option_fields:
+                params['conditions'].append({prop: value})
+            else:
+                params[prop] = value
+
+        return self._service_api_connection.service_request_async(self.service, params, use_json=True)
+
+    @class_or_instance
+    def query_criteria_async(self, **criteria):
+        """
+        Given a set of search criteria, returns a list of mission metadata.
+
+        Parameters
+        ----------
+        **criteria
+            Criteria to apply. At least one non-positional criteria must be supplied.
+            Valid criteria are coordinates, objectname, radius (as in `query_region` and `query_object`),
+            and all fields listed in the column documentation for the mission being queried.
+            Fields that can be used to match results on criteria. See the TAP schema link below for all field names.
+            https://vao.stsci.edu/missionmast/tapservice.aspx/tables#folder38
```

Comment:
This is not a requirement for the PR, but @tomdonaldson, I wonder maybe there is a human-readable version of this (or a link to a tool that does that, with that XML preloaded into it)? (e.g the workflow of astroquery users to look at the XML of that link to dig up field names feels a bit clunky to me)

Reply:
I totally agree with this comment.  Unfortunately we don't support a clean human-readable format for this, and I'd be hesitant to point to a 3rd-party formatter.

@syed-gilani it's not obvious to me what the `#folder38` part of the URL does.  Are there environments where that would limit what parts of the TAP Schema are shown?